docs/CNVs/CNV-methods.tex

```diff
@@ -829,44 +916,99 @@ \subsection{Generative Model for Read Counts} \label{coverage-model}
 
-In exact mode the leading cost is a few terms of $O(D^2 T S)$ flops, each with small prefactors, per iteration.  In approximate mode it is $O(D T S)$ flops.  Assuming a total prefactor of $10$ and $T = 2 \times 10^5$, $D = 10$, $S = 500$ a full EM iteration costs $10^{11}$ flops in exact mode and $10^{10}$ in approximate mode.  On a single 1 GHz core ($10^9$ flops per second) this comes out to roughly 100 or 10 seconds.  Even in exact mode we will probably get most of the way toward convergence with approximate iterations, followed by several exact iterations.  Very conservatively, 100 approximate iterations and 10 exact ones should suffice, which comes out to less than half an hour.
+\noindent {\bf (Idea 1) Blind source separation ---} One remedy is to use a blind source separation approach, such as independent component analysis (ICA), to separate the signal from the noise as a first step, followed by learning the latent features of the noise using PCA. In ICA-like methods, one decomposes the signal into additive subcomponents and minimizes the mutual information between them (or maximizes the non-Gaussanity by taking into account higher moments such as the kurtosis). Even though this method is quite appealing, we follow a more context-specific heuristic approach here.\\
 
-We can apply the parameters learned from the panel of normals to single-sample calling, which requires the likelihood as a function of the copy numbers $\vc$.  Applying the same E step as above, the likelihood is
+\noindent {\bf (Idea 2) Imposing context-specific constraints on the structure of variations ---} Fortunately, we have some idea about the spatial structure of the CNV events: they are tandem amplification or attenuation of the read count over several targets. In the absence of noise, we expect the frequency spectrum of the CNV signal (as obtained by taking a Fourier transform of $m_{st}$ in $t$) to be significantly enhanced at spatial frequencies corresponding to the inverse length scale of the size of the CNV event. Similarly, the variation subtracted from sample $s$, i.e. $\vW \vz_s$, will exhibit an enhanced spectral power if a CNV event is erroneously picked up. Let $\tilde{f}(k)$ be the Fourier transform of a linear filter that approximately represents a range of CNV events. For example, we may use a midpass filter such as:
+\begin{equation}
+\tilde{f}(k) = \left\{
+\begin{array}{ll}
+  1 & \quad k_l \leq k^* \leq k_h,\\
+  \\
+  0 & \quad k^* > k_h \,\, \text{or} \,\, k^* < k_l,
+\end{array}
+\right.
+\end{equation}
+Here, $k^* = \min(k, T - 1 - k)$, $k_l \sim \lfloor T / \ell_\mathrm{max} \rfloor$ and $k_l \sim \lfloor T / \ell_\mathrm{min} \rfloor$, where $\ell_\mathrm{min}$ and $\ell_\mathrm{max}$ denote roughly the minimum and maximum length of the CNV events in the units of targets. The filtered spectral power of the noise in sample $s$ is given as:
```

Comment:
In principle, the best strategy is to (1) make a new matrix $W_{t\mu} \rightarrow W_{x\mu}$ with $x$ representing the genomic position, and $W_{x\mu} \equiv W_{t(x)\mu}$ where $t(x)$ is the target that overlaps with $x$, and (2) perform Fourier filtering on $W_{x\mu}$. If target spacing is not too non-uniform, this might be unnecessarily extra work. This is easy to code but lets to -extremely- large matrices and time-consuming FFT operations. If the target spacing is not terribly non-uniform, we may leave it as it is. What do you guys think? @davidbenjamin @samuelklee